### PR TITLE
Add support for JavaScript+React Admin UIs

### PR DESCRIPTION
**Pull request by AlCalzone**
**PR Checklist:**

-   [x] Provide a meaningful description to this PR or mention which issues this fixes.
-   [x] Add tests for your change. This includes negative tests (i.e. inputs that need to fail) as well as baseline tests (i.e. how should the directory structure look like?).
-   [x] Run the test suite with `npm test`
-   [x] If there are baseline changes, review them and make a separate commit for them with the comment "accept baselines" if they are desired changes
-   [x] Ensure the project builds with `npm run build`
-   [ ] ~~If you added a required option, also add it to the template creation (`.github/create_templates.ts`)~~
-   [ ] Add a detailed migration description to `docs/updates` explaining what the user needs to do when manually updating an existing project
-   [x] Add your changes to `CHANGELOG.md` (referencing the migration description and this PR or the issue you fixed)

**Description:**  
This PR adds support for building React Admin UIs with JavaScript

No migration guide this time, the changes are too complex. Just create a new adapter and copy over the normal adapter files

### CODE REVIEW

File: templates/admin/src/components/settings.jsx.ts
Comment (UncleSamSwiss) [inline templates/admin/src/components/settings.jsx.ts]: I would suggest to use one template for JSX and TSX, there are only 8 different sections and most of the code is the same.

Comment (UncleSamSwiss) [reply templates/admin/src/components/settings.jsx.ts]: ... similar to index.tsx_jsx.ts

File: templates/admin/src/app.raw.jsx
Comment (UncleSamSwiss) [inline templates/admin/src/app.raw.jsx]: I would suggest to use one template for JSX and TSX similar to index.tsx_jsx.ts. We won't be able to use `.raw` anymore, but that's not that bad.

**Review by UncleSamSwiss — COMMENTED** [review]
A few comments related to similar code between JSX and TSX.

**Comment by GermanBluefox** [comment]
Ready for merge?

**Comment by AlCalzone** [comment]
Waiting for a last review from @UncleSamSwiss

**Review by UncleSamSwiss — APPROVED** [review]
Reviewed the latest changes. Looks good to me.

**Pull request merged**
